Cannot build GlassFish 7.0.13-SNAPSHOT (6611a36875a4fe8be16a9e5eddaa433af7c6c465) with JDK 11

Building GlassFish 7.0.13-SNAPSHOT (6611a36875a4fe8be16a9e5eddaa433af7c6c465) with JDK11 does not seem to work. When I am building with JDK 11, I get a problem that I don't seem to get with JDK 17 (I will redact certain path aspects with the sequence _REDACT_ - hopefully the general issue isn't masked by that):
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:3.1.0:run (do stuff) on project glassfish: An Ant BuildException has occured: The following error occurred while executing this line:
[ERROR] _REDACT_/glassfish/appserver/distributions/glassfish/target/antrun/build-main.xml:9: The archive microprofile-jwt-auth-api.jar doesn't exist
[ERROR] around Ant part ...<jarupdate basedir="_REDACT_/glassfish/appserver/distributions/glassfish/src/main/patches/microprofile-jwt-auth-api" destfile="_REDACT_/glassfish/appserver/distributions/glassfish/target/stage/glassfish7/glassfish/modules/microprofile-jwt-auth-api.jar" includes="META-INF/MANIFEST.MF" />... @ 16:318 in _REDACT_/glassfish/appserver/distributions/glassfish/target/antrun/build-main.xml
```
I think it should be able to work, and the README indicates it to be expected to work: https://github.com/eclipse-ee4j/glassfish/blob/6611a36875a4fe8be16a9e5eddaa433af7c6c465/README.md?plain=1#L10

--- a/appserver/distributions/glassfish/pom.xml
+++ b/appserver/distributions/glassfish/pom.xml
@@ -81,6 +81,24 @@
 
             <plugin>
                 <artifactId>maven-antrun-plugin</artifactId>
+                <dependencies>
+                    <dependency>
+                        <groupId>org.apache.ant</groupId>
+                        <artifactId>ant</artifactId>
+                        <version>${ant.version}</version>
+                    </dependency>
+                    <dependency>
+                        <groupId>ant-contrib</groupId>
+                        <artifactId>ant-contrib</artifactId>
+                        <version>1.0b2</version>
+                        <exclusions>
+                            <exclusion>
+                                <groupId>ant</groupId>
+                                <artifactId>ant</artifactId>
+                            </exclusion>
+                        </exclusions>
+                    </dependency>
+                </dependencies>
                 <executions>
                     <execution>
                         <id>do stuff</id>
@@ -89,12 +107,24 @@
                             <goal>run</goal>
                         </goals>
                         <configuration>
-                            <target>
+                            <target xmlns:if="ant:if" xmlns:unless="ant:unless">
+                                <taskdef resource="net/sf/antcontrib/antcontrib.properties" classpathref="maven.plugin.classpath" />
+                                
+                                <if>
+                                    <bool>
+                                        <isgreaterthan arg1="${ant.java.version}" arg2="16"/>
+                                    </bool>
+                                    <then>
+                                        <property name="java17orGreater" value="true"/>
+                                    </then>
+                                </if>
+                                
                                 <macrodef name="jarupdate">
                                     <attribute name="basedir"/>
                                     <attribute name="includes"/>
                                     <attribute name="destfile"/>
                                     <sequential>
+                                        <echo>PATCHING</echo>
                                         <zip destfile="@{destfile}.tmp">
                                             <zipfileset src="@{destfile}" excludes="@{includes}"/>
                                         </zip>
@@ -102,11 +132,12 @@
                                         <zip update="true" basedir="@{basedir}" includes="@{includes}" destfile="@{destfile}" />
                                     </sequential>
                                 </macrodef>
-                                <jarupdate
+                                
+                                <jarupdate if:set="java17orGreater"
                                     basedir="${patches}/microprofile-jwt-auth-api" includes="META-INF/MANIFEST.MF"
                                     destfile="${glassfish.modules}/microprofile-jwt-auth-api.jar"
                                 />
-                                <jarupdate
+                                <jarupdate if:set="java17orGreater"
                                     basedir="${patches}/microprofile-rest-client-api" includes="META-INF/MANIFEST.MF"
                                     destfile="${glassfish.modules}/microprofile-rest-client-api.jar"
                                 />

